Ghost-CLI: npm install sqlite3 --save error

Currently when using ghost-cli 1.0.0.4 and using `ghost install local` during the installation I get this  error and the installation fails. 

![image](https://user-images.githubusercontent.com/18176012/27876943-7f5c6e42-617e-11e7-9a15-6dea7422927a.png)

I tried to remove everything (purge nodejs from system and remove all node_modules across system), so this is like a fresh install and I've still got the error.

```
$ npm ls -gp --depth=0 | awk -F/ '/node_modules/ && !/\/npm$/ {print $NF}' | xargs npm -g rm
$ sudo apt-get remove --purge nodejs
$ sudo apt-get install nodejs
$ npm install -g ghost-cli
```

Ubuntu 16.04
Node v6.11.0

```
➜  blog ghost install local
✔ Checking for latest Ghost version
✔ Running system checks
✔ Setting up install directory
✔ Downloading and installing Ghost v1.0.0-beta.2
✔ Moving files
✔ Configuring Ghost
ℹ Running setup checks [skipped]
✔ Setting up instance
Knex: run
$ npm install sqlite3 --save
Error: Cannot find module '/usr/lib/node_modules/ghost-cli/node_modules/sqlite3/lib/binding/node-v48-linux-x64/node_sqlite3.node'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/lib/node_modules/ghost-cli/node_modules/sqlite3/lib/sqlite3.js:4:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Client_SQLite3._driver (/usr/lib/node_modules/ghost-cli/node_modules/knex/lib/dialects/sqlite3/index.js:75:12)
    at Client_SQLite3.initializeDriver (/usr/lib/node_modules/ghost-cli/node_modules/knex/lib/client.js:219:26)
    at Client_SQLite3.Client (/usr/lib/node_modules/ghost-cli/node_modules/knex/lib/client.js:113:10)
    at new Client_SQLite3 (/usr/lib/node_modules/ghost-cli/node_modules/knex/lib/dialects/sqlite3/index.js:58:20)
```

--- a/lib/commands/doctor/checks/install.js
+++ b/lib/commands/doctor/checks/install.js
@@ -17,7 +17,7 @@ const eol = os.EOL;
 
 const tasks = {
     // While it's not an actual task, we put it here to make it easier to test
-    checkDirectoryAndAbove: function checkDirectoryAndAbove(dir) {
+    checkDirectoryAndAbove: function checkDirectoryAndAbove(dir, extra) {
         if (isRoot(dir)) {
             return Promise.resolve();
         }
@@ -29,14 +29,24 @@ const tasks = {
                 return Promise.reject(new errors.SystemError(
                     `The path ${dir} is not readable by other users on the system.${eol}` +
                     'This can cause issues with the CLI, please either make this directory ' +
-                    'readable by others or install in another location.'
+                    `readable by others or ${extra} in another location.`
                 ));
             }
 
-            return checkDirectoryAndAbove(path.join(dir, '../'));
+            return tasks.checkDirectoryAndAbove(path.join(dir, '../'), extra);
         });
     },
-    nodeVersion: function nodeVersion() {
+    nodeVersion: function nodeVersion(ctx) {
+        let globalBin = execa.shellSync('npm bin -g').stdout;
+
+        if (process.argv[1] !== path.join(__dirname, '../../../../bin/ghost') && !process.argv[1].startsWith(globalBin)) {
+            return Promise.reject(new errors.SystemError(
+                `The version of Ghost-CLI you are running was not installed with this version of Node.${eol}` +
+                `This means there are likely two versions of Node running on your system, please ensure${eol}` +
+                'that you are only running one global version of Node before continuing.'
+            ));
+        }
+
         if (process.env.GHOST_NODE_VERSION_CHECK !== 'false' &&
             !semver.satisfies(process.versions.node, cliPackage.engines.node)) {
             return Promise.reject(new errors.SystemError(
@@ -48,7 +58,11 @@ const tasks = {
             ));
         }
 
-        return Promise.resolve();
+        if (ctx.local || os.platform() !== 'linux' || (ctx.argv && ctx.argv['setup-linux-user'] === false)) {
+            return Promise.resolve();
+        }
+
+        return tasks.checkDirectoryAndAbove(process.argv[0], 'install node and Ghost-CLI');
     },
     folderPermissions: function folderPermissions(ctx) {
         return fs.access(process.cwd(), constants.R_OK | constants.W_OK).catch(() => {
@@ -61,7 +75,7 @@ const tasks = {
                 return Promise.resolve();
             }
 
-            return tasks.checkDirectoryAndAbove(process.cwd());
+            return tasks.checkDirectoryAndAbove(process.cwd(), 'run `ghost install`');
         });
     },
     systemStack: function systemStack(ctx) {

--- a/test/unit/commands/doctor/install-spec.js
+++ b/test/unit/commands/doctor/install-spec.js
@@ -3,6 +3,7 @@ const expect = require('chai').expect;
 const sinon = require('sinon');
 const stripAnsi = require('strip-ansi');
 const proxyquire = require('proxyquire').noCallThru();
+const path = require('path');
 
 const modulePath = '../../../../lib/commands/doctor/checks/install';
 const errors = require('../../../../lib/errors');
@@ -34,47 +35,38 @@ describe('Unit: Doctor Checks > Install', function () {
     });
 
     describe('node version check', function () {
-        it('doesn\'t do anything if GHOST_NODE_VERSION_CHECK is false', function () {
-            let originalEnv = process.env;
-            let cliPackage = {
-                engines: {
-                    node: '0.10.0'
-                }
-            };
-            process.env = { GHOST_NODE_VERSION_CHECK: 'false' };
+        it('rejects if global bin is different than the one ghost is running from', function () {
+            let execaStub = sinon.stub().returns({stdout: '/usr/local/bin'});
+            let originalArgv = process.argv;
+            process.argv = ['node', '/home/ghost/.nvm/versions/6.11.1/bin'];
 
             const task = proxyquire(modulePath, {
-                '../../../../package': cliPackage
+                execa: {shellSync: execaStub}
             }).tasks.nodeVersion;
 
             return task().then(() => {
-                process.env = originalEnv;
+                expect(false, 'error should be thrown').to.be.true;
+                process.argv = originalArgv;
+            }).catch((error) => {
+                process.argv = originalArgv;
+                expect(error).to.be.an.instanceof(errors.SystemError);
+                expect(error.message).to.match(/version of Ghost-CLI you are running was not installed with this version of Node./);
+                expect(execaStub.calledOnce).to.be.true;
+                expect(execaStub.calledWithExactly('npm bin -g')).to.be.true;
             });
         });
 
-        it('doesn\'t do anything if node version is in range', function () {
-            let cliPackage = {
-                engines: {
-                    node: process.versions.node // this future-proofs the test
-                }
-            };
-
-            const task = proxyquire(modulePath, {
-                '../../../../package': cliPackage
-            }).tasks.nodeVersion;
-
-            return task();
-        });
-
-        it('throws error if node version is not in range', function () {
+        it('rejects if node version is not in range', function () {
             let cliPackage = {
                 engines: {
                     node: '0.10.0'
                 }
             };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
 
             const task = proxyquire(modulePath, {
-                '../../../../package': cliPackage
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub}
             }).tasks.nodeVersion;
 
             return task().then(() => {
@@ -85,6 +77,140 @@ describe('Unit: Doctor Checks > Install', function () {
 
                 expect(message).to.match(/Supported: 0.10.0/);
                 expect(message).to.match(new RegExp(`Installed: ${process.versions.node}`));
+                expect(execaStub.calledOnce).to.be.true;
+            });
+        });
+
+        it('doesn\'t reject if bin is the local ghost bin file from the install (and local is true)', function () {
+            let cliPackage = {
+                engines: {
+                    node: process.versions.node // this future-proofs the test
+                }
+            };
+            let execaStub = sinon.stub().returns({stdout: '/usr/local/bin'});
+            let originalArgv = process.argv;
+            process.argv = ['node', path.join(__dirname, '../../../../bin/ghost')];
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: true}).then(() => {
+                process.argv = originalArgv;
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.called).to.be.false;
+            });
+        });
+
+        it('doesn\'t do anything if GHOST_NODE_VERSION_CHECK is false and local is true', function () {
+            let originalEnv = process.env;
+            let cliPackage = {
+                engines: {
+                    node: '0.10.0'
+                }
+            };
+            process.env = { GHOST_NODE_VERSION_CHECK: 'false' };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: true}).then(() => {
+                process.env = originalEnv;
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.called).to.be.false;
+            });
+        });
+
+        it('doesn\'t do anything if node version is in range and local is true', function () {
+            let cliPackage = {
+                engines: {
+                    node: process.versions.node // this future-proofs the test
+                }
+            };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: true}).then(() => {
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.called).to.be.false;
+            });
+        });
+
+        it('doesn\'t call checkDirectoryAndAbove if os is not linux', function () {
+            let cliPackage = {
+                engines: {
+                    node: process.versions.node // this future-proofs the test
+                }
+            };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
+            let platformStub = sinon.stub().returns('darwin');
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub},
+                os: {platform: platformStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: false}).then(() => {
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.called).to.be.false;
+            });
+        });
+
+        it('doesn\'t call checkDirectoryAndAbove if no-setup-linux-user is passed', function () {
+            let cliPackage = {
+                engines: {
+                    node: process.versions.node // this future-proofs the test
+                }
+            };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
+            let platformStub = sinon.stub().returns('linux');
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub},
+                os: {platform: platformStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: false, argv: {'setup-linux-user': false}}).then(() => {
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.called).to.be.false;
+            });
+        });
+
+        it('calls checkDirectoryAndAbove if none of the three conditions are true', function () {
+            let cliPackage = {
+                engines: {
+                    node: process.versions.node // this future-proofs the test
+                }
+            };
+            let execaStub = sinon.stub().returns({stdout: process.argv[1]});
+            let platformStub = sinon.stub().returns('linux');
+
+            const tasks = proxyquire(modulePath, {
+                '../../../../package': cliPackage,
+                execa: {shellSync: execaStub},
+                os: {platform: platformStub}
+            }).tasks;
+            let checkDirectoryStub = sinon.stub(tasks, 'checkDirectoryAndAbove').resolves();
+
+            return tasks.nodeVersion({local: false, argv: {'setup-linux-user': true}}).then(() => {
+                expect(execaStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.calledOnce).to.be.true;
+                expect(checkDirectoryStub.calledWith(process.argv[0])).to.be.true;
             });
         });
     });

